Pre-processing of CWL workflows and tools for conformance tests on PAPI

Horneth opened the pull request:
I removed the default gcs prefix and default docker conf thinking that they only existed for conformance tests but maybe we want to keep them for other purposes ?

cjllanwarne commented on centaurCwlRunner/src/main/scala/centaur/cwl/PAPIPreprocessor.scala:
```diff
@@ -0,0 +1,121 @@
+  private def prefixLocationWithGcs(key: String): Json => Json = root.selectDynamic(key).string.modify(gcsPrefix + _)
+
+  // Prefix "location" and "path"
+  private val prefix = prefixLocationWithGcs("location").compose(prefixLocationWithGcs("path"))
```
We shouldn't ever need to change `path` to be GCS. That's for the localized path on the VM (vs the universal resource **Location** being a fully qualified, accessible path)

kshakir replied:
The above comment is already an open ticket if someone wants to pick it up: #3132

kshakir commented on centaurCwlRunner/src/main/scala/centaur/cwl/PAPIPreprocessor.scala:
```diff
@@ -0,0 +1,121 @@
+/**
+  * Tools to pre-process the CWL workflows and inputs before feeding them to Cromwell so they can be executed on PAPI.
+  */
+object PAPIPreprocessor {
```
ToL: Awesome. I'll refactor this into a generic processor that can feed in a prefix jic we want to test other backends too.

Horneth commented on centaurCwlRunner/src/main/scala/centaur/cwl/PAPIPreprocessor.scala:
```diff
@@ -0,0 +1,121 @@
+  */
+object PAPIPreprocessor {
+  // GCS directory where inputs for conformance tests are stored
+  private val gcsPrefix = "gs://centaur-cwl-conformance/cwl-inputs/"
```
editors and owners of our (Cromwell dev) google project

Horneth commented:
@cjllanwarne I removed the `path` prefixing and added tests for `location` in `default`.
Also added the file prefixing to CWL workflows (was only for input files before). If @danbills's PR on cwltool gets merged and we can salad CWL workflows with gcs path it should unlock at least another test I believe.

mcovarr commented on centaurCwlRunner/src/main/scala/centaur/cwl/CentaurCwlRunner.scala:
```diff
@@ -113,8 +115,8 @@ object CentaurCwlRunner extends StrictLogging {
-    val workflowContents = parsedWorkflowPath.contentAsString
-    val inputContents = args.workflowInputs.map(_.contentAsString)
+    val workflowContents = preProcessIfNotLocal(PAPIPreprocessor.preProcessWorkflow)(parsedWorkflowPath.contentAsString)
+    val inputContents = args.workflowInputs.map(_.contentAsString) map preProcessIfNotLocal(PAPIPreprocessor.preProcessInput)
```
ugh oops nm 😦
